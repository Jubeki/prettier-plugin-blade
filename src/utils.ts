import { format } from "prettier";
// @ts-ignore
import php from "@prettier/plugin-php/standalone";

export const formatAsPhp = (source: string): string => {
  if (!source.startsWith("<?php")) {
    source = "<?php " + source;
  }

  let code = format(source, { parser: "php", plugins: [php] })
    .replace("<?php ", "")
    .trim();

  if (source.trim().endsWith(";")) {
    return code;
  }

  // The PHP plugin for Prettier will add a semi-colon by default. We don't always want that.
  return code.substring(0, code.length - 1);
};
