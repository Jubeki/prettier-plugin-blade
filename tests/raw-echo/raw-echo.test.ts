import fs from "fs";
import path from "path";
import { format } from "../utils";

const fixture = fs.readFileSync(path.join(__dirname, "./raw-echo.blade.php"), "utf-8");

test("basic", () => {
    const formatted = format(fixture);
    expect(formatted).toMatchSnapshot();
});
