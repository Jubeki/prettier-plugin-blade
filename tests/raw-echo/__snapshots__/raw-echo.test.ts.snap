// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
"<h1>Here's some very poorly formatted {!! $rawEchos !!}</h1>
"
`;
